src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +112,36 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    IEdmStructuralProperty structuralProperty = null;
+                    if (!string.IsNullOrEmpty(keyProperty?.Alias))
+                    {
+                        string[] keyPropertySegments = keyProperty.PropertyName.Split('/');

```

Comment:
Because we are the ones creating this string and not the client - I think we should take care of all this things at the point of creating the string when writing to the csdl.